client/client.go

```diff
@@ -220,6 +220,12 @@ func (c *Client) Connect(config *Config) (ConnectFuture, error) {
 		connect.Username = urlParts.User.Username()
 		connect.Password, _ = urlParts.User.Password()
 	}
+	if config.Username != "" {
```

Comment:
Because the security level of password and endpoint (broker URL) are different and our projects store all passwords encrypted, it is not easy to join the password and endpoint (broker URL). So I hope it can support config password separately.

Reply:
Ok I understand. But then we should do this properly and remove the URL from the `Config` struct and unpack the pieces earlier. That way there is no overriding happening. I would appreciate if we could do this in a separate PR and remove the change for now.